src/proxy/inbound.rs

```diff
@@ -235,7 +235,9 @@ impl Inbound {
         };
         let illegal_call = if pi.cfg.inpod_enabled {
             // User sent a request to pod:15006. This would forward to pod:15006 infinitely
-            illegal_ports.contains(&upstream_addr.port())
+            // Use hbone_addr instead of upstream_addr to allow for sandwich mode, which intentionally
+            // sends to 15008.
+            illegal_ports.contains(&hbone_addr.port())
```

Comment:
is there any test to make sure it won't break again?

Reply:
I am working on it currently, it requires a large refactor of our test setup. I will be done tomorrow